src/AbstractProxyFixer.php

```diff
@@ -28,15 +28,17 @@
 abstract class AbstractProxyFixer extends AbstractFixer
 {
     /**
-     * @var array<string, FixerInterface>
+     * @var non-empty-array<string, FixerInterface>
      */
-    protected array $proxyFixers = [];
+    protected array $proxyFixers;
 
     public function __construct()
     {
+        $proxyFixers = [];
         foreach (Utils::sortFixers($this->createProxyFixers()) as $proxyFixer) {
-            $this->proxyFixers[$proxyFixer->getName()] = $proxyFixer;
+            $proxyFixers[$proxyFixer->getName()] = $proxyFixer;
         }
+        $this->proxyFixers = $proxyFixers;
```

Comment:
@keradus (ref https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/9010/files#r2315699409) 
`$this->createProxyFixers()` returns non-empty-list. 
With template in `Utils::sortFixers` phpstan knows that `Utils::sortFixers(non-empty-list)` also returns `non-empty-list`.
So phpstan knows that `$proxyFixers` is non-empty-list. So no error here when assigning `$proxyFixers` to non-empty-list property `$this->proxyFixers`.

Reply:
sounds quirky for me to need this flip-flop ,
but not such prio to waste discussion on it. let's move on